src/libraries/System.Text.Json/gen/JsonSourceGenerator.Emitter.cs

```diff
@@ -1322,7 +1322,11 @@ private string GetGetTypeInfoImplementation()
 @$"/// <inheritdoc/>
 public override {JsonTypeInfoTypeRef}? GetTypeInfo({TypeTypeRef} type)
 {{");
-
+                // This method body grows linearly over the number of generated types.
+                // In line with https://github.com/dotnet/runtime/issues/77897 we should
+                // eventually replace this method with a direct call to Options.GetTypeInfo().
+                // We can't do this currently because Options.GetTypeInfo throws whereas
+                // this GetTypeInfo returns null for unsupported types, so we need new API to support it.
```

Comment:
And we either don't want to or can't catch the exception?  Just wondering about the tradeoffs involved, how common it is for it to throw, etc.

Reply:
It throws if attempting to resolve unsupported types. I considered catching the exception but it seemed like a hack.